pkg/skaffold/deploy/resource/deployment.go

```diff
@@ -54,11 +54,14 @@ func (d *Deployment) Deadline() time.Duration {
 
 func (d *Deployment) UpdateStatus(details string, err error) {
 	updated := newStatus(details, err)
-	if !d.status.Equal(updated) {
-		d.status = updated
-		if strings.Contains(details, rollOutSuccess) || isErrAndNotRetryAble(err) {
-			d.done = true
-		}
+	if d.status.Equal(updated) {
+		d.status.changed = false
+		return
```

Comment:
Is this return appropriate here?  You still need to check the `isErrAndNotRetryable(err)` to possibly set `done`?

Reply:
umm no. the status is already same as previous state. We don't need to perform any checks.